typescript/aws-transfer-sftp-server/aws-transfer-sftp-server.ts

```diff
@@ -0,0 +1,152 @@
+import { Stack, StackProps, RemovalPolicy } from 'aws-cdk-lib';
+import {
+  aws_s3 as s3,
+  aws_ec2 as ec2,
+  aws_iam as iam,
+  aws_logs as logs,
+  aws_cloudwatch as cw,
+  aws_transfer as transfer,
+} from 'aws-cdk-lib';
+import { Construct } from 'constructs';
+
+/** Properties required to setup the SFTP server. */
+export interface SftpServerStackProps extends StackProps {
+  /** User name for the default user.
+   * @example 'sftp-user'
+   */
+  userName: string;
+  /** The public key of the user */
+  userPublicKeys: string[];
+  /** Allowed IPs. 
+   * @default All IPV4 is allowed.
+   */
+  allowedIps?: string[];
+}
+
+/** Stack for initializing a fully working SFTP server. */
+export class SftpServerStack extends Stack {
+
+  /** S3 bucket for storing the incoming data. */
+  incomingDataBucket: s3.Bucket;
+
+  /** CloudWatch alarm that is triggered if there are too many errors in the logs. */
+  errorAlarm: cw.Alarm;
+
+  constructor(scope: Construct, id: string, props: SftpServerStackProps) {
+    super(scope, id, props);
+
+    const vpc = new ec2.Vpc(this, 'VPC', {
+      maxAzs: 2,
+      natGateways: 0,
+    });
+
+    this.incomingDataBucket = new s3.Bucket(this, 'IncomingDataBucket', {
+      blockPublicAccess: s3.BlockPublicAccess.BLOCK_ALL,
+      bucketName: `sftp-server-data-bucket-${this.account}-${this.region}`,
+      encryption: s3.BucketEncryption.KMS_MANAGED,
+      enforceSSL: true,
+      // Do not use for production
```

Comment:
Don't put it in an example. 

This bucket should be passed up through the `SftpServerStackProps`.